crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -856,20 +861,40 @@ impl OlmMachine {
         transaction: &mut StoreTransaction,
         event: &EncryptedToDeviceEvent,
         changes: &mut Changes,
-    ) -> OlmResult<OlmDecryptionInfo> {
+    ) -> OlmResult<Option<OlmDecryptionInfo>> {
+        // Decrypt the event
         let mut decrypted =
             transaction.account().await?.decrypt_to_device_event(&self.inner.store, event).await?;
 
-        // We ignore all to-device events from dehydrated devices - we should not
-        // receive any
-        if !self.to_device_event_is_from_dehydrated_device(&decrypted, &event.sender).await? {
-            // Handle the decrypted event, e.g. fetch out Megolm sessions out of
-            // the event.
+        let from_dehydrated_device = self
+            .to_device_event_is_from_dehydrated_device(&decrypted, &event.sender)
+            .await
+            .map_err(|e| {
+                error!(
+                    error = ?e,
+                    "Couldn't check whether the event is from a dehydrated device",
+                );
+                e
+            })?;
```

Comment:
Why logging the error if the error is returned? Is the error not accurate enough?